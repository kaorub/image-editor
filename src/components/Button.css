.Button {
    margin: 8px;
    color: rgba(0, 0, 0, 0.87);
    box-shadow: 0 3px 1px -2px rgba(0,0,0,0.2), 0 2px 2px 0 rgba(0,0,0,0.14), 0 1px 5px 0 rgba(0,0,0,0.12);
    background-color: rgb(144, 202, 249);
    border-color: rgba(0, 0, 0, 0.87);
    color: #fff;
    padding: 6px 16px;
    font-size: 0.875rem;
    min-width: 64px;
    box-sizing: border-box;
    transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    font-weight: 500;
    line-height: 1.75;
    border-radius: 4px;
    letter-spacing: 0.02857em;
    text-transform: uppercase;
    width: 50%;
}
