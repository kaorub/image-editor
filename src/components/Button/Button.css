.Button {
    margin: 8px;
    background-color: rgb(144, 202, 249);
    color: #fff;
    padding: 6px 16px;
    min-width: 64px;
    box-sizing: border-box;
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    line-height: 1.75;
    border-radius: 4px;
    border-width: 0;
    text-transform: uppercase;
    width: 50%;
}
