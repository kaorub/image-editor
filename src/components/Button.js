/**
 * Button component to render button
 * @property name unique name of button
 * @property value
 * @property label - displayed name
 */
import React from 'react';
import './Button.css';

function Button(props) {
  return <div>
    <button name={props.name} value={props.value} className="Button">
      {props.label}
    </button>
  </div>
}
export default Button;
