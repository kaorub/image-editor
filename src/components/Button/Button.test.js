import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';
import renderer, { create } from 'react-test-renderer';

it('renders button component', () => {
  const btn = shallow(<Button />);
  expect(btn.find('.Button').length).toEqual(1);
  expect(btn.find('div').length).toEqual(1);
});

it('checks button props', () => {
  const btn = shallow(<Button label="test"/>);
  expect(btn.text()).toEqual("test");
});

it('checks button clicked event', () => {
  const fn = jest.fn();
  let tree = create(<Button onClick={fn} />);
  const button = tree.root.findByType('button');
  button.props.onClick()
  expect(fn.mock.calls.length).toBe(1);
});

it('compares button snapshot', () => {
  const fn = jest.fn();
  const component = renderer.create(
      <Button onClick={fn}/>,
  );
  let btn = component.toJSON();
  expect(btn).toMatchSnapshot();
});
