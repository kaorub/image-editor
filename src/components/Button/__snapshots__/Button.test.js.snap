// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compares button snapshot 1`] = `
<div>
  <button
    className="Button"
    onClick={[MockFunction]}
  />
</div>
`;
